usl_models/tests/model_test.py

```diff
@@ -183,3 +184,38 @@ def test_bad_labels():
 
     with pytest.raises(AssertionError):
         model.train([fake_data])
+
+
+def test_early_stopping():
+    """Tests early stopping during model training.
+
+    Since inputs are random, the model will not continually improve (there's no real
+    training). Setting a reasonably large number of epochs makes it likely that the
+    validation loss stops improving before training is complete. Enabling early stopping
+    with a patience of 1 epoch means that the model should stop training as soon as the
+    validaton loss goes up.
+
+    Note that this test is nondeterministic and possibly flaky. But with a large enough
```

Comment:
If we set a random seed such that we're consistently using the same "random" inputs every time, I think that would address possible flakiness?
https://www.tensorflow.org/api_docs/python/tf/random/set_seed

Reply:
You're right! Thanks for the suggestion, added a seed.